sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,282 @@
+    2016-06-06 00:00:00 2016-06-07 00:00:00   GONG Data not Available
+    2016-06-07 00:00:00 2016-06-08 00:00:00   GONG Data not Available
+    2016-06-08 00:00:00 2016-06-09 00:00:00   GONG Data not Available
+    2016-06-09 00:00:00 2016-06-10 00:00:00   GONG Data not Available
+    2016-06-10 00:00:00 2016-06-11 00:00:00   GONG Data not Available
+    2016-06-11 00:00:00 2016-06-12 00:00:00   GONG Data not Available
+    2016-06-12 00:00:00 2016-06-13 00:00:00   GONG Data not Available
+    2016-06-13 00:00:00 2016-06-14 00:00:00   GONG Data not Available
+                    ...                 ...    ...                ...
+    2016-06-26 00:00:00 2016-06-27 00:00:00   GONG Data not Available
+    2016-06-27 00:00:00 2016-06-28 00:00:00   GONG Data not Available
+    2016-06-28 00:00:00 2016-06-29 00:00:00   GONG Data not Available
+    2016-06-29 00:00:00 2016-06-30 00:00:00   GONG Data not Available
+    2016-06-30 00:00:00 2016-07-01 00:00:00   GONG Data not Available
+    2016-07-01 00:00:00 2016-07-02 00:00:00   GONG Data not Available
+    2016-07-02 00:00:00 2016-07-03 00:00:00   GONG Data not Available
+    2016-07-03 00:00:00 2016-07-04 00:00:00   GONG Data not Available
+    2016-07-04 00:00:00 2016-07-05 00:00:00   GONG Data not Available
+    2016-07-05 00:00:00 2016-07-06 00:00:00   GONG Data not Available]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        #TO-DO: Figure out where scraper would and wouldn't work
+        table_physobs = {'INTENSITY' : 'i', 'LOS_MAGNETIC_FIELD': 'b'} #legitimate physical observations.
+        table_instruments = ['bb', 'ct', 'le', 'ml', 'td', 'ud','z'] #For magnetogram and intensity
+        
+        physobs_in = True if ('physobs' in kwargs.keys() and kwargs['physobs'] in table_physobs.keys()) else False #Is PhysObs entered
+        instrument_in = True if ('instrument' in kwargs.keys() and kwargs['instrument'] in table_instruments) else False
+        wavelength_in = True if 'wavelength' in kwargs.keys() else False
+        #Is instrument entered.
+        url_pattern_1 =  'ftp://gong2.nso.edu/QR/{id}qa/%Y%m/{ObsID}bqa%y%m%d/{ObsID}bqa%y%m%dt%H%M.fits.gz'
+        url_pattern_2 = 'http://gong2.nso.edu/HA/haf/%Y%m/%Y%m%d/%Y%m%d%H%M%S{ObsID}h.fits.fz'
+        
+        result = list() #final list of urls
+        patterns = list()
+        if not physobs_in:
+            patterns.append(url_pattern_1), patterns.append(url_pattern_2)
+        else:
+            if kwargs['physobs'] == 'LOS_MAGNETIC_FIELD':
+                patterns.append(url_pattern_1)
+            else:
+                #Differentiate on basis of wavelength
+                if not wavelength_in:
+                    patterns.append(url_pattern_1), patterns.append(url_pattern_2)
+                else:
+                    wave = int(kwargs['wavelength'].min.value)
```

Comment:
I had written about this on #1761 , but I guess it got lost in all notifications. I will repost this,

Hey, @dpshelio . I have been playing with ```astropy.units```. Its pretty cool. And I have tried converting different units from one unit to the other. It works but there are floating point issues. 
In our case the ```Wavelength``` class in ```sunpy.vso.attrs``` automatically converts any compatible units to Angstroms (Its in the docs). So we explicitly worry about converting units to Angstroms. But the main issue is **floating point conversion**. It doesn't get converted into the exact value that we want. 
There are two ways you can get around this.

- Define an epsilon and if difference b/w our value and desired value is less than epsilon its fine and we can accept the value.
- Round-off the value after conversion.

I think the first one is better (I'm pretty familiar with this).